### ISSUE TITLE
Conflict with orgstruct-mode

### ISSUE THREAD

**Issue by drveera** [kyleam/snakemake-mode#27]
Hi, 
I use orgstruct mode to fold and expand sections of the script. When I try the same with snake file, it works as long as the snakemake-mode is not activated, but then it stops working after I activate the snakemake-mode. Is it because of some conflict?

I structured the code like this 

```
#!/bin/env snakemake

### * imports
### * outfiles
### * rules

### * org mode sepcific
### Local Variables:
### eva: (orgstruct-mode 1)
### orgstruct-heading-prefix-regexp: "### "
### End:
```

Could you please help figure out why orgstruct-mode stops working, when I activate snakemake-mode ?

**Comment by kyleam** [comment]
Thanks for the information.  I've never used orgstruct-mode and don't know how it intercepts the commands. I probably won't get around to looking into this until this weekend.  In the meantime

  * What version of Emacs and Org are you using?

    (Note that orgstruct-mode was removed in Org 9.1.6, so regardless of whether this issue is resolved here, you'll probably want to look into alternatives.)

  * Does orgstruct-mode work for you with in python-mode (the one defined python.el and shipped with Emacs)?

**Comment by kyleam** [comment]
I've not been able to trigger the issue.  I've tried using the tip of Org's maint branch, release_9.1.7-19-gea4cb1488 (as I mentioned above, orgstruct-mode is removed in master, which will be Org 9.2).

Right, so I have a Snakefile that looks like this (note the s/eva/eval/ to fix the typo in your original snippet):

```
### * imports

from glob import glob

### * rules

rule blah:
    output: "foo"
    shell: "echo blah > {output}"

### * org mode sepcific
### Local Variables:
### eval: (orgstruct-mode 1)
### orgstruct-heading-prefix-regexp: "### "
### End:
```

With Snakemake mode and Org on the load path and no additional configuration (i.e., `emacs -Q`), if I visit that file and hit tab on the imports line, it folds as expected.

**Comment by drveera** [comment]
Hi
Very sorry for the late response. Thanks a lot for taking time to look at this. After seeing your comments, I did explore it a little. The problem was I manually activated snakemake-mode and while doing so I lost the orgstruct mode (I assume), which is the reason why it didn't work for me. Now when I added the add-to-list in my configuration file for snakemake-mode. Now when I open the snakemake file i get both modes active and the folding works!! Thanks for your help.
